docs/docs/why-gatsby-uses-graphql.md

```diff
@@ -0,0 +1,360 @@
+
+Test this query by entering it into the left-hand panel of the GraphQL Playground, then pressing the play button in the top center.
+
+The results will appear in the panel between the query and the docs, and they’ll look like this:
+
+![GraphQL Playground](./images/why-gql-playground.png)
+
+### Generate pages with GraphQL
+
+In `gatsby-node.js`, we can use the GraphQL query we just wrote to generate pages.
+
+```js:title=gatsby-node.js
+exports.createPages = async ({ actions: { createPage }, graphql }) => {
+  const results = await graphql(`
+    {
+      allProductsJson {
+        edges {
+          node {
+            slug
+          }
+        }
+      }
+    }
+  `)
+
+  results.data.allProductsJson.edges.forEach(edge => {
+    const product = edge.node
+
+    createPage({
+      path: `/gql/${product.slug}/`,
+      component: require.resolve("./src/templates/product-graphql.js"),
+      context: {
+        slug: product.slug,
+      },
+    })
+  })
+}
+```
+
+You need to use the `graphql` helper that’s available to the [`createPages` API](/docs/node-apis/#createPages) to execute the query. To make sure that the result of the query comes back before continuing, use [`async`/`await`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function).
+
+The results that come back are very similar to the contents of `data/products.json`, so you can loop through the results and create a page for each.
+
+However, note that you’re only passing the `slug` in `context` — you’ll use this in the template component to load more product data.
+
+As you’ve already seen, the `context` argument is made available to the template component in the `pageContext` prop. To make queries more powerful, Gatsby _also_ exposes everything in `context` as a GraphQL variable, which means you can write a query that says, in plain English, “Load data for the product with the slug passed in `context`.”
+
+Here’s what that looks like in practice:
+
+```jsx:title=src/templates/product-graphql.js
+import React from "react"
+import { graphql } from "gatsby"
+import Image from "gatsby-image"
+
+export const query = graphql`
+  query($slug: String!) {
+    productsJson(slug: { eq: $slug }) {
+      title
+      description
+      price
+      image {
+        childImageSharp {
+          fluid {
+            ...GatsbyImageSharpFluid
+          }
+        }
+      }
+    }
+  }
+`
+
+const Product = ({ data }) => {
+  const product = data.productsJson
+
+  return (
+    <div>
+      <h1>{product.title}</h1>
+      <Image
+        fluid={product.image.childImageSharp.fluid}
+        alt={product.title}
+        style={{ float: "left", marginRight: "1rem", width: 150 }}
+      />
+      <p>{product.price}</p>
+      <div dangerouslySetInnerHTML={{ __html: product.description }} />
+    </div>
+  )
+}
+
+export default Product
+```
+
+A few notes about this file:
+
+1. The result of the query is added to the template component as the `data` prop.
```

Comment:
```suggestion
1. The result of the query is injected to the template component as the `data` prop.
```

Reply:
I went back and forth on some of this wording. My intention is to avoid any computer science-ey phrasing to make sure we don't lose people on syntax. I know that's a losing battle, but here we are. 😄

Reply:
Leave it :) It _feels like_ we generally use injected (so that's why I suggested) but maybe that term is in itself confusing?

Reply:
I think that it's one of those things that you don't know until you know. It's hard for me to be sure whether that's me being paranoid or if it's a valid concern, though.